streamdeck_ui/api.py

```diff
@@ -699,3 +751,41 @@ def _update_button_filters(self, serial_number: str, page: int, button: int):
             )
 
         display_handler.replace(page, button, filters)
+
+    def prepare_plugin(self, serial_number: str, page_id: int, button_id: int, plugin_path: str) -> Union[Plugin, None]:
+        plugin = None
+        plugin_config = self.get_button_plugin_config(serial_number, page_id, button_id)
+        try:
+            full_path = os.path.expanduser(os.path.expandvars(plugin_path))
+            result = runpy.run_path(full_path)
+            for name, obj in result.items():
+                if inspect.isclass(obj) and issubclass(obj, Plugin) and hasattr(obj, 'initialize_plugin') and callable(
+                        getattr(obj, 'initialize_plugin')) and obj != Plugin:
+                    plugin = obj.initialize_plugin(serial_number, page_id, button_id, plugin_config)
+
+                    # Connects event functions to the api
+                    plugin.change_text_callback = self.on_update_button_text
```

Comment:
since button states and the ext of each one. is more like logic inside the plugin that something that we need to store what do you think of prepare alternative method than do not save nothing to the .json and just calls the device updates

Reply:
I dont fully understand what you mean by that, could you maybe explain it again?
How I understand it you want the plugin to not be saved in each button state?

Reply:
Would the current implementation attempt to overwrite the plugin's button state modifications if we don't overwrite the json?